napari/layers/_vectors_layer/model.py

```diff
@@ -443,13 +332,26 @@ def _set_view_slice(self):
         """Sets the view given the indices to slice with."""
 
         vertices = self._mesh_vertices
-        faces = self._mesh_triangles
+
+        if len(self.data) == 0:
+            faces = []
+        elif self.ndim > 2:
+            matches = np.all(self.indices[:-2] == self.data[:, 0, :-2], axis=1)
+            matches = np.where(matches)[0]
+            if len(matches) == 0:
+                faces = []
+            else:
+                keep_inds = np.repeat(2 * matches, 2)
+                keep_inds[1::2] = keep_inds[1::2] + 1
+                faces = self._mesh_triangles[keep_inds]
+        else:
+            faces = self._mesh_triangles
 
         if len(faces) == 0:
             self._node.set_data(vertices=None, faces=None)
         else:
             self._node.set_data(
-                vertices=vertices[:, ::-1], faces=faces, color=self.color
+                vertices=vertices[:, ::-1] + 0.5, faces=faces, color=self.color
```

Comment:
I think this is related to the cursor status bug I reported in #310, and probably the wrong way to deal with it...? ie you are curing the symptom here instead of the cause (that the image pixels are offset).

Reply:
:point_up: this is the serious one LOL, sorry for the vague comment!

Reply:
Yeah, so this is also happening in the `Points` layer here https://github.com/napari/napari/blob/961cb2cad10aab7228aa1cec8a443a7e5bd004df/napari/layers/_points_layer/model.py#L386
and is likely related to the issue in #310, but I'd much rather systematically fix this in a follow-up PR addressing just this stuff then try and add this on here. It will also affect the painting in the labels layer and the addition of polygons in the shapes layer